src/auro-datepicker.js

```diff
@@ -182,49 +176,54 @@ class AuroDatePicker extends LitElement {
   }
 
   /**
-   * Determines the element error state based on the `required` attribute and input value.
+   * Determines the validity state of the element.
    * @private
    * @returns {void}
    */
-  handleRequired() {
-    if (this.required && !this.error) {
-      if (!this.triggerInput.value || this.triggerInput.value.length === 0) {
-        this.error = true;
-        this.setAttribute('error', '');
-      } else {
-        this.error = false;
-        this.removeAttribute('error');
-      }
-    }
-  }
-
-  /**
-   * Validates if the selected date is within defined range.
-   * @private
-   * @returns {void}
-   */
-  validateDate() {
-    let error = false;
+  validate() {
+    this.validity = 'valid';
+    this.input.setCustomValidity = '';
 
-    if (this.minDate) {
-      if (new Date(this.minDate) > new Date(this.input.value)) {
-        error = true;
-      }
-    }
-
-    if (this.maxDate) {
-      if (new Date(this.maxDate) < new Date(this.input.value)) {
-        error = true;
+    /**
+     * Only validate once we interact with the datepicker
+     * this.value === undefined is the initial state pre-interaction.
+     *
+     * The validityState definitions are located at https://developer.mozilla.org/en-US/docs/Web/API/ValidityState.
+     */
+    const validDateStrLength = 10;
+    if (this.value !== undefined) {
+      if ((!this.input.value || this.input.value.length === 0) && this.required) {
+        this.validity = 'valueMissing';
+        this.input.setCustomValidity = 'Please enter a date.';
+      } else {
+        const date = new Date(this.input.value);
+
+        if (this.input.value.length > 0 && !this.validDate(date)) {
+          this.validity = 'badInput';
```

Comment:
Is the idea here that the user would be injecting the validity state or this is all internal? When I am using the demo page, I am unable to get any of these error statements to appear.

Reply:
Datepicker (more accurately calendar) has current restrictions that prevent you from selecting a date before the minDate or after the maxDate. So it is impossible to use the datepicker normally and make those errors appear. If that restriction were removed from Calendar the errors can be made to show. 

Today, the only errors you can make happen are for the required flag and having no value (after first interaction) and tooShort (start typing a date but don't finish and then blur the input)